dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitor.java

```diff
@@ -0,0 +1,145 @@
+    private final SystemMessageEventUtil systemMessageEventUtil;
+
+    private AtomicBoolean started = new AtomicBoolean();
+    private final ScheduledThreadPoolExecutor scheduledThreadPoolExecutor;
+
+    private Timer timer;
+
+    private ESReadOnlyMonitor(final SystemMessageEventUtil systemMessageEventUtil, final RoleAPI roleAPI) {
+        this.systemMessageEventUtil = systemMessageEventUtil;
+        this.roleAPI = roleAPI;
+        started.set(false);
+
+        scheduledThreadPoolExecutor = new ScheduledThreadPoolExecutor(1);
+        scheduledThreadPoolExecutor.setContinueExistingPeriodicTasksAfterShutdownPolicy(true);
+    }
+
+    private ESReadOnlyMonitor() {
+        this(SystemMessageEventUtil.getInstance(), APILocator.getRoleAPI());
+    }
+
+    // Inner class to provide instance of class
+    private static class Singleton
+    {
+        private static final ESReadOnlyMonitor INSTANCE = new ESReadOnlyMonitor();
+    }
+
+    @VisibleForTesting
+    public static ESReadOnlyMonitor getInstance(
+            final SystemMessageEventUtil systemMessageEventUtil,
+            final RoleAPI roleAPI)
+    {
+        return new ESReadOnlyMonitor(systemMessageEventUtil, roleAPI);
+    }
+    public static ESReadOnlyMonitor getInstance()
+    {
+        return Singleton.INSTANCE;
+    }
+
+    public void start(final ReindexEntry reindexEntry, final String cause){
+        if (started.compareAndSet(false, true)) {
+            Logger.error(this.getClass(), Thread.currentThread().getName() + " - Reindex failed for :" + reindexEntry + " because " + cause);
+
+            if (ESIndexUtil.isAnyCurrentIndicesReadOnly()) {
+                ReindexThread.setCurrentIndexReadOnly(true);
+                sendMessage("es.index.read.only.message");
+                startMonitor();
+            } else {
+                started.set(false);
+            }
+        }
+    }
+
+    private void sendMessage(final String messageKey) {
+        try {
+            final Role adminRole = roleAPI.loadCMSAdminRole();
+            final List<String> usersId = roleAPI.findUsersForRole(adminRole)
+                    .stream()
+                    .map(user -> user.getUserId())
+                    .collect(Collectors.toList());
+
+            final SystemMessageBuilder message = new SystemMessageBuilder()
+                    .setMessage(LanguageUtil.get(messageKey))
+                    .setSeverity(MessageSeverity.ERROR)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setLife(TimeUnit.SECONDS.toMillis(5));
+
+            systemMessageEventUtil.pushMessage(message.create(), usersId);
+        } catch (final LanguageException | DotDataException | DotSecurityException e) {
+            Logger.warn(ESReadOnlyMonitor.class, () -> e.getMessage());
+        }
+    }
+
+    private void putCurrentIndicesToWriteMode() {
+        try {
+            Logger.debug(this.getClass(), "Trying to set the current indices to Write mode");
```

Comment:
remember use the supplier

Reply:
done https://github.com/dotCMS/core/pull/18604/commits/fc979e0e1d04475951dbb213b379b17022474a8e#diff-8528ec811f60c9aae6301f9071d59872R102